corehq/apps/hqwebapp/templatetags/hq_shared_tags.py

```diff
@@ -625,6 +625,13 @@ def render(self, context):
     return AddToBlockNode(nodelist, 'js-inline')
 
 
+@register.simple_tag
+def html_attr(value):
+    if not isinstance(value, basestring):
+        value = JSON(value)
+    return escape(value)
```

Comment:
jQuery's `.data()` figures it out:

> When the data attribute is an object (starts with '{') or array (starts with '[') then jQuery.parseJSON is used to parse the string; it must follow valid JSON syntax including quoted property names. If the value isn't parseable as a JavaScript value, it is left as a string.

In rather un-javascript-like fashion, it also decides whether the value is a string versus a number:
> Every attempt is made to convert the string to a JavaScript value (this includes booleans, numbers, objects, arrays, and null). A value is only converted to a number if doing so doesn't change the value's representation. For example, "1E02" and "100.000" are equivalent as numbers (numeric value 100) but converting them would alter their representation so they are left as strings. The string value "100" is converted to the number 100.

(from https://api.jquery.com/data/)